### PR TITLE
add extract-text-webpack-plugin helper package

### PR DESCRIPTION
**Pull request by erquhart**
Provides shared per-stage `extract-text-webpack-plugin` instances for all Gatsby plugins, including Gatsby core (implementation PR will follow). Gatsby styling plugins currently overwrite one another's output during the build whenever multiple `extract-text-webpack-plugin` instances are added to the Webpack config (the Netlify CMS plugin does this). This happens because the plugins are importing `extract-text-webpack-plugin` and instantiating a new instance direct from the constructor. The module added by this PR allows those plugins to import the same instance used by all other plugins.

In short, this enables a simpler and more reliable way to achieve the fix from #4495 for all plugins.

Related to #4516.

Note: the conventions in this PR are borrowed from `gatsby-1-config-css-modules` with the assumption that that package is a desirable pattern for Gatsby config helpers.

### CODE REVIEW

**Comment by erquhart** [comment]
@KyleAMathews this PR will include the package for initial publishing, after that I'll put up a PR for implementing it in Gatsby's webpack config and the Sass plugin. We can give that a bit of time in case bugs surface before implementing in the remaining statically extracted style plugins.

**Comment by KyleAMathews** [comment]
Nice! Will merge and publish so you can start on the implementation. Thanks for jumping into this!
